### ISSUE TITLE
On Android BGRA texture quality is lower than YCbCr420_OES

### ISSUE THREAD

**Issue by Last8Exile** [RenderHeads/UnityPlugin-AVProVideo#1818]
**Describe the issue**
On Android BGRA texture quality is lower than YCbCr420_OES.
Texture appears blocky and pixelated on object edges. Bilinear filtering when sampling BGRA texure not helping and pixelation still visible.
On other platforms (Mac, iOS, visionOS) BGRA texure quality is good and the same as YCbCr420_OES. On Windows texture quality also the same.
As a workaround I request YCbCr420_OES texure and resolve it to RenderTexture using ResolveToRenderTexture component.

Documentation only mention that YCbCr420_OES is faster and requre less memory. No mentions of difference in quality.

**Your Setup:**
 - Unity version: 2022.3.22f1
 - AVPro Video version: 3.0.0f1-ultra, 2.7.3f1-ultra
 - Operating system version: Android 14, Android 12
 - Device model: Oculus Quest 2 and 3, RedMagic 9, Google Pixel 6a
 - Video specs (resolution, frame-rate, codec, file size): 4320x2160, 59.940 FPS, mp4 HEVC YUV 4:2:0 8 bit, 1Mb

**To Reproduce**
1. Compare top-middle part of test.mp4 video on Android. Edges of red mask will be pixelated when using BGRA texture and smooth when using YCbCr420_OES texture.
2. On other platforms edges will be smooth for both texture settings.

**Expected**
I expect same (smooth) quality with both BGRA and YCbCr420_OES textures on Android, as it is on other platforms.

**Screenshots**
[test_bgra.png](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/8383387/63b32571-cfd1-4151-a44b-fd2e1d2bf6fb) - exact texture content from Android using BGRA texture. Blocky.
[test_oes.png](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/8383387/a481e9c9-13de-410a-8d3b-5b97eaa7c380) - exact texture content from Android using YCbCr420_OES texture resolved to RenderTexture. Smooth.
[test_point.png](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/8383387/6cbdc543-cf3c-41fd-98ef-064e1f177f20) - side by side comparison of test_bgra.png and test_oes.png using point filtering. BRGA side looks blocky.
[test_bilinear.png](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/8383387/49f88a25-44dd-4335-a350-8b320f269a4d) - side by side comparison of test_bgra.png and test_oes.png using bilinear filtering. BRGA side looks pixelated even after filtering.

**Videos**
[test.mp4](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/8383387/46a36107-c688-4b3b-aad2-d22cd8f5e94a) - video created from static image. Top-middle part contains red mask. Edges on mask looks blocky only when using BGRA texture on Android.

**Comment by MorrisRH** [comment]
Some sampler state was stale, will be fixed in the next release.

**Comment by Chris-RH** [comment]
AVPro Video version 3.0.1 has been released.  Please let us know if it has not fixed the issue for you.

**Comment by Last8Exile** [comment]
Not fixed. BGRA setting on Android still produces pixelated textures and YCbCr420_OES is smooth.

I reverted back to 2.7.3
On Android there is BlitTextureFiltering setting (which is deprecated in later versions).
With Point it produces same pixelated texture.
With Bilinear it produces smooth texture. Same as with OES + ResolveToRenderTexture and same as on other platforms with BGRA textures.

May be this is intended behaviour. I never compared raw OES texure to BGRA texture (because I can't adapt OES shader to my needs). I only compared BGRA to OES + ResolveToRenderTexture.

Seems on all platforms BGRA textures use bilinear filterening at some point and only on Android it uses point filtering.
On 2.7.3 I have control over that with BlitTextureFiltering setting and don't need to use OES to get smooth texture.
